pkg/protocol/engine/blocks/block.go

```diff
@@ -19,8 +19,8 @@ type Block struct {
 	// BlockDAG block
 	missing             bool
 	missingBlockID      iotago.BlockID
-	solid               bool
```

Comment:
if it's a bool variable it could be an event, then you can also call `Trigger` on it without having to pass value - that way you can get rid of the SetSolid methods I think

Reply:
Considering that a reactive.Event is a more complex type, as it that enapsulates a readable.Variable, and considering that a block is so ubiquitous, I would like to keep a simpler API rather than having `b.Booked().WasTriggered()` and such. It seems less readable to me!